### ISSUE TITLE
Partition table does not support autorandom feature

### ISSUE THREAD

**Issue by zhenjiaogao** [pingcap/tidb#19784]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
MySQL [gzj_test]> CREATE TABLE `partition_tab` (
    -> `id` bigint(20) unsigned NOT NULL auto_random,
    -> `create_date` date NOT NULL,
    -> PRIMARY KEY (`id`,`create_date`)
    -> )
    -> PARTITION BY RANGE COLUMNS(create_date) (
    -> PARTITION p20191215 VALUES LESS THAN ("20191216"),
    -> PARTITION p20191216 VALUES LESS THAN ("20191217")
    -> );
ERROR 8216 (HY000): Invalid auto random: column id is not the integer primary key, or table is created with alter-primary-key enabled
MySQL [gzj_test]>
MySQL [gzj_test]> show config where type='tidb' and name like '%alter-primary-key%';
+------+-------------------+-------------------+-------+
| Type | Instance          | Name              | Value |
+------+-------------------+-------------------+-------+
| tidb | 100.10.10.10:4568 | alter-primary-key | false |
+------+-------------------+-------------------+-------+
1 row in set (0.05 sec)
```

### 2. What did you expect to see? (Required)
Partition table  supports autorandom feature.

### 4. What is your TiDB version? (Required)

```sql
MySQL [gzj_test]> select version();
+--------------------+
| version()          |
+--------------------+
| 5.7.25-TiDB-v4.0.4 |
+--------------------+
1 row in set (0.00 sec)
```

**Comment by zz-jason** [comment]
@bb7133 Can we regard `auto_random` as the brother property of `auto_increment`? So that whenever `auto_increment` is supported, `auto_random` should also be supported.

**Comment by tangenta** [comment]
> PRIMARY KEY (`id`,`create_date`)

For now, `AUTO_RANDOM` requires that the primary key contains exactly 1 column.

**Comment by sultan8252** [comment]
Add a scene description：
In large-scale log record storage scenarios (200 million records per day), records older than 60 days need to be cleaned up, and the AUTO_RANDOM function is required to avoid writing and query hot issues

**Comment by yahonda** [comment]
I think this issue can be closed because TiDB 6.4 implements it.

```sql
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.4.0
Edition: Community
Git Commit Hash: cf36a9ce2fe1039db3cf3444d51930b887df18a1
Git Branch: heads/refs/tags/v6.4.0
UTC Build Time: 2022-11-13 05:15:26
GoVersion: go1.19.2
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
1 row in set (0.00 sec)

mysql> use test;
Database changed
mysql> CREATE TABLE `partition_tab` (
    -> `id` bigint(20) unsigned NOT NULL auto_random,
    -> `create_date` date NOT NULL,
    -> PRIMARY KEY (`id`,`create_date`)
    -> )
    -> PARTITION BY RANGE COLUMNS(create_date) (
    -> PARTITION p20191215 VALUES LESS THAN ("20191216"),
    -> PARTITION p20191216 VALUES LESS THAN ("20191217")
    -> );
Query OK, 0 rows affected, 1 warning (0.13 sec)

mysql> show warnings;
+-------+------+---------------------------------------------------------+
| Level | Code | Message                                                 |
+-------+------+---------------------------------------------------------+
| Note  | 1105 | Available implicit allocation times: 576460752303423487 |
+-------+------+---------------------------------------------------------+
1 row in set (0.00 sec)


mysql> show create table partition_tab\G
*************************** 1. row ***************************
       Table: partition_tab
Create Table: CREATE TABLE `partition_tab` (
  `id` bigint(20) unsigned NOT NULL /*T![auto_rand] AUTO_RANDOM(5) */,
  `create_date` date NOT NULL,
  PRIMARY KEY (`id`,`create_date`) /*T![clustered_index] CLUSTERED */
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY RANGE COLUMNS(`create_date`)
(PARTITION `p20191215` VALUES LESS THAN ('20191216'),
 PARTITION `p20191216` VALUES LESS THAN ('20191217'))
1 row in set (0.00 sec)
```

**Comment by tangenta** [comment]
closed by #38617.
